### PR TITLE
Add cuda12 build support [skip ci]

### LINKED ISSUE #7843: [FEA] build with CUDA 12
**Is your feature request related to a problem? Please describe.**
Speed up the build process of spark-rapids to support cuda11 and cuda12.

**Describe the solution you'd like**
We're planning to support CUDA 12 build. So, we'll have a classifier "-cuda12" by packaging the spark-rapids-jni jar built for cuda 12. I suppose spark-rapids-jni will also have a classifier "-cuda12".
The scala/java codes in spark-rapids are not related to cuda 12. So, it doesn't need to build twice.
The difference between cuda11 and cuda12 will be only the native libraries, which are in spark-rapids-jni jar.
For now, we only package the spark-rapids-jni jar in the stage of building dist folder.

The ideal process of building spark-rapids for cuda11 and cuda12 will be:
1. Build the jars for the sub modules(except dist).
2. Package the dist with the sub modules and spark-rapids0-jni-cuda11 jar
3. Package the dist with the sub modules and spark-rapids0-jni-cuda12 jar

**Describe alternatives you've considered**
Without this, we simply can build the jars of sub modules twice but waste some time.

### PR DESCRIPTION
**Pull request by pxLi**
fix #7843 

Update CI nightly build script to support multiple cuda classifiers.
The idea here is to keep existing process as before, 
and all cuda-classifier dists build would share the same pre-built aggregators.

I choose to move non-default-classifier artifacts back to dist/target folder before deploy is because that it would be easier for updating existing release process after nightly build.

Verified locally for build, test, deployment. If this implementation looks OK, I will make corresponding changes to our internal pipeline. Also merging this PR would not take effect before we add `CUDA_CLASSIFIERS=cuda11,cuda12` to our nightly pipeline

### CODE REVIEW

File: jenkins/spark-nightly-build.sh [hunk 1]
```diff
@@ -27,12 +27,16 @@ MVN="mvn -Dmaven.wagon.http.retryHandler.count=3 -DretryFailedDeploymentCount=3
...
 DIST_PL="dist"
 function mvnEval {
-    $MVN help:evaluate -q -pl $DIST_PL $MVN_URM_MIRROR -Prelease311 -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER -DforceStdout -Dexpression=$1
+    $MVN help:evaluate -q -pl $DIST_PL $MVN_URM_MIRROR -Prelease311 -Dmaven.repo.local=$M2DIR -DforceStdout -Dexpression=$1
 }
 
 ART_ID=$(mvnEval project.artifactId)
 ART_GROUP_ID=$(mvnEval project.groupId)
 ART_VER=$(mvnEval project.version)
+CUDA_CLASSIFIER=${CUDA_CLASSIFIER:-$(mvnEval cuda.version)} # default cuda version
```
Comment (tgravescs) [inline jenkins/spark-nightly-build.sh:36]: can we call this DEFAULT_CUDA_CLASSIFIER?

**Review by tgravescs — COMMENTED** [review]
>  and all cuda-classifier dists build would share the same pre-built aggregators.

I'm not sure what you mean by this?

File: jenkins/spark-nightly-build.sh (hunk 1 above)
Comment (pxLi) [reply jenkins/spark-nightly-build.sh:36]: yes, just need to update pipeline ENV. I will rename it

**Comment by tgravescs** [comment]
> > > and all cuda-classifier dists build would share the same pre-built aggregators.
> > 
> > 
> > I'm not sure what you mean by this?
> 
> The requirement is at #7843, the goal is not to build all spark shims again for another cuda classifier to save resources and time

So I assume thi sis referring to: The scala/java codes in spark-rapids are not related to cuda 12. So, it doesn't need to build twice.  ?

Maybe I'm missing it, this is doing a mvn clean install on everything again, so it is going to rebuild everything, right? or at this point is DIST_PROFILE_OPT set to only do the dist module?

**Review by tgravescs — APPROVED** [review]

**Comment by pxLi** [comment]
> So I assume thi sis referring to: The scala/java codes in spark-rapids are not related to cuda 12. So, it doesn't need to build twice. ?
> Maybe I'm missing it, this is doing a mvn clean install on everything again, so it is going to rebuild everything, right? or at this point is DIST_PROFILE_OPT set to only do the dist module?

yes, the cuda12 dist install is trying to build dist module w/ jni-cuda12 artifact. And yes, it would build other modules for default spark, 311 here, but we would only keep the dist one for later deployment. And for the following cuda11 w/ default 311 build, it will actually deploy the intermediate modules (like sql, aggregator, tests, etc) of 311 to URM.

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
let me merge this one and verify update to internal pipeline.
If any issues, I will deal w/ in follow-ups, thanks!

**Pull request merged**